src/main/java/com/nha/abdm/wrapper/hip/hrp/database/mongo/services/PatientService.java

```diff
@@ -88,4 +93,64 @@ public boolean checkCareContexts(InitResponse data) {
     }
     return true;
   }
+
+  public AckRequest addPatient(LinkRecordsResponse data) {
+    String abhaAddress = data.getAbhaAddress();
+    try {
+      Patient existingRecord = this.patientRepo.findByAbhaAddress(abhaAddress);
+      if (existingRecord == null) {
+        throw new RuntimeException("Patient not found");
+      } else {
+        List<Map<String, Object>> modifiedCareContexts =
+            data.getPatient().getCareContexts().stream()
+                .map(
+                    careContext -> {
+                      Map<String, Object> modifiedContext = new HashMap<>();
+                      modifiedContext.put("referenceNumber", careContext.getReferenceNumber());
+                      modifiedContext.put("display", careContext.getDisplay());
+                      modifiedContext.put("isLinked", false);
+                      return modifiedContext;
+                    })
+                .collect(Collectors.toList());
+        Query query = new Query(Criteria.where("abhaAddress").is(data.getAbhaAddress()));
+        Update update = new Update().addToSet("careContext").each(modifiedCareContexts); // TODO
```

Comment:
what does this TODO mean? what is pending here?

Reply:
Nothing in TODO, modified this PR according to the previous PR's TODO